library/core/src/panicking.rs

```diff
@@ -217,16 +228,17 @@ fn assert_failed_inner(
 
     match args {
         Some(args) => panic!(
-            r#"assertion failed: `(left {} right)`
+            r#"assertion failed: `({left_name} {} {right_name})`
   left: `{:?}`,
```

Comment:
The `at: ` here shouldn't be followed by a formatting specifier that we print, this is meant to be where the location goes, which is printed by the caller of `assert_failed_inner`, so we should just finish the output here. The `args` that you're printing here is the error message and should come before `left`. 

```suggestion
            r#"assertion failed: `({left_name} {op} {right_name})`
Error: {args},
 left: `{left_val:?}`,
```